File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/UnitOfWorkEventRecord.cs
using System;
using System.Collections.Generic;

namespace Volo.Abp.Uow;

public class UnitOfWorkEventRecord
{
    public Object EventData;//  { get; }

    public Type EventType;//  { get; }

    public long EventOrder;//  { get; }

    public boolean UseOutbox;//  { get; }

    /**
     * Extra properties can be used if needed.
    */
    public Dictionary<String, Object> Properties;//  { get; } = new Dictionary<string, object>();

    public UnitOfWorkEventRecord(
        Type eventType,
        Object eventData,
        long eventOrder,
        boolean useOutbox = true)
    {
        EventType = eventType;
        EventData = eventData;
        EventOrder = eventOrder;
        UseOutbox = useOutbox;
    }
}

File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/UnitOfWorkEventRecord.java
package com.ueueo.uow;

import lombok.Getter;

import java.util.HashMap;
import java.util.Map;

@Getter
public class UnitOfWorkEventRecord {
    private Object eventData;

    private Class<?> eventType;

    private long eventOrder;

    private boolean useOutbox;

    /**
     * Extra properties can be used if needed.
     */
    private Map<String, Object> Properties = new HashMap<>();

    public UnitOfWorkEventRecord(
            Class<?> eventType,
            Object eventData,
            long eventOrder,
            Boolean useOutbox)
    {
        this.eventType = eventType;
        this.eventData = eventData;
        this.eventOrder = eventOrder;
        this.useOutbox = useOutbox != null ? useOutbox : true;
    }
}
